src/lib/normalise-dag-node.js

```diff
@@ -10,20 +70,48 @@ import { toCidOrNull, getCodecOrNull } from './cid'
  *
  * ^: currently dag-cbor and dag-pb are supported.
  *
+ * @function normaliseDagNode
  * @param {Object} node the `value` prop from `ipfs.dag.get` response.
- * @param {String} cid the cid string passed to `ipfs.dag.get`
+ * @param {string} cidStr the cid string passed to `ipfs.dag.get`
  */
 export default function normaliseDagNode (node, cidStr) {
-  const type = getCodecOrNull(cidStr)
-  if (type === 'dag-pb') {
-    return normaliseDagPb(node, cidStr, type)
+  const code = getCodeOrNull(cidStr)
+  if (code === dagPb.code) {
+    return normaliseDagPb(node, cidStr, code)
   }
   // try cbor style if we don't know any better
-  return normaliseDagCbor(node, cidStr, type)
+  return normaliseDagCbor(node, cidStr, code ?? dagCbor.code)
 }
 
-/*
+/**
+ *
+ * @param {import('@ipld/dag-pb').PBNode} node
+ * @returns {DagPbNodeAsJson}
+ */
+export function convertDagPbNodeToJson (node) {
+  const pbData = /** @type {Uint8Array} */(node.Data)
+  let data = String.fromCharCode(...pbData)
+  try {
+    data = JSON.parse(data)
+  } catch {
+    // ignore
+  }
+
+  return {
+    type: node.Links.length > 0 ? 'directory' : 'file',
```

Comment:
was this logic always in here? I'm not sure this is a very safe assumption, you kind of need to dig into the bytes to figure out what's in there; this heuristic might be _mostly_ correct, but probably safe enough to bank on. You could bring in unixfs to decode the dag-pb `Data` field further. js-unixfs is a newer implementation of unixfs that might be worth a look at, you could use the `decode` there, or perhaps figure out an export of the `Data` encoder from there (or ask for it to be exported if it's not): https://github.com/ipld/js-unixfs/blob/main/src/codec.js#L412-L414 - that would give you an accurate `type` if it's necessary.

Reply:
oh, UnixFS from ipfs-unixfs is already here, so you could just use that yo do this work

Reply:
This is newer code that was essentially created in order to replace the `node = node.toJSON()` call inside of the `normaliseDagPb` function.

How do I convert `import('@ipld/dag-pb').PBNode` to a json object? 
Do I need to? 
What exactly is normal for `normaliseDagPb`? 
is `DagPbNodeAsJson` typed sanely?

Reply:
Do we have any types defined anywhere that make it easy to see what the LCT and GCT (least and greatest common type; not sure what the name for this is in type theory) are for dagPb + json ? What about dagPb and Cbor? what about cbor and json? What about other object types? 

Is this repo/pkg the only place where we have examples of converting between these objects? Does it make sense to encapsulate this logic elsewhere? is there an ipld/convert package with a `normalizedIpld` object that all datatypes be converted to so we can easy translate a dagPb node into a dagCbor node? 

```
type dagPb2NormIpld = (node: DagPbNode) => NormIpldNode
type dagCbor2NormIpld = (node: DagCbor) => NormIpldNode
type dagJson2NormIpld  = (node: DagJSON) => NormIpldNode
type normIpld2DagPb = (node: NormIpldNode) => DagPbNode
type normIpld2DagCbor = (node: NormIpldNode) => DagCbor
type normIpld2DagJson = (node: NormIpldNode) => DagJSON
```

Reply:
Ok. I've removed the new function `convertDagPbNodeToJson` entirely and no longer need to depend on it.